src/pages/index.js

```diff
@@ -74,19 +35,57 @@ const validationOptions = {
 
 const editFormValidator = new FormValidator(validationOptions, profileEditForm);
 const addCardFormValidator = new FormValidator(validationOptions, addCardForm);
+const profilePictureValidator = new FormValidator(
+  validationOptions,
+  profilePictureForm
+);
 
 editFormValidator.enableValidation();
 addCardFormValidator.enableValidation();
+profilePictureValidator.enableValidation();
 
 // Card
 
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", () =>
-    handleImageClick(cardData)
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    toggleLike
   );
   section.addItem(card.getView());
 }
 
+function toggleLike(card) {
+  if (card.isLiked) {
+    api
+      .removeLikesReq(card.id)
+      .then(() => {
+        card.handleLikeIcon(false);
+      })
+      .catch((err) => console.error(err));
+  } else {
+    api
+      .addLikesReq(card.id)
+      .then(() => {
+        card.handleLikeIcon(true);
+      })
+      .catch((err) => console.error(err));
+  }
+}
+
+function handleDeleteClick(card) {
+  confirmationModal.open();
+  confirmationModal.setSubmitAction(() => {
+    api.deleteCardReq(card.id).then(() => {
+      card.remove();
+    });
+  });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**